### PR TITLE
Adapt simulator facts to shared revised HSL reconstruction

### PR DESCRIPTION
**Pull request by enarjord**
Revised HSL needs the simulator to supply the same factual inputs as live reconstruction. Add an isolated adapter from actual simulator fill records, current positions, native contract metadata, raw balance and candle closes to the shared price/snapshot/trace components. Preserve execution sequence, gross PnL and signed fees; ignore derived fill after-states. Keep configured versus dynamic-tradability slot counts explicit, and include both sides in unified scope regardless of entry enablement.

Value completed 1m candles at their closing timestamp, retain simulator fill timestamps and explicit candle/fill phase and global execution order, and clip both histories to the configured window. Retain a flat delisted pair's in-window history using its last factual close with a diagnostic; a fresh mark remains required for any held exposure, and balance/positions remain fresh. This prevents a now-irrelevant quote from blocking reconstruction of a flat scope's cooldown. An explicitly requested flat coin with no retained fills carries fresh coin/side/window-bound proof, producing a neutral trace without fabricating a quote. Bare absence remains an error.

This is a staged input adapter, not revised trading activation. Existing live/backtest/optimizer guards remain in place and legacy stays the default. The final sparse-history dispatcher, runtime application, metrics and full revised fake-live integration remain subsequent work.

Validation: 328 Rust tests, default-feature compile checks, 2,542 Python tests and documentation checks passed with a rebuilt/source-verified extension. New tests drive the simulator's real fill handlers through shared reconstruction for both sides, partial closes, units/fees, same-bar cross-pair flatten/reopen, pre-entry candle ordering, quote-free confirmed flat scopes, unified disabled-side exposure, raw balance, dynamic slots, delisting, inclusive lookback edges retaining the actual boundary close, source-column mapping and future-price exclusion. All validation is offline.

### CODE REVIEW

**Comment by enarjord** [comment]
@codex review

Please independently review the simulator-input adapter at 6f55794061a6c70a616f1a59c9c44f2bd1fcb9c1, particularly event versus candle-close timing, native units and execution sequence, in-window boundaries, scope/slot semantics and the stale-mark exception restricted to confirmed-flat pairs. This PR supplies shared reconstruction inputs but deliberately retains all revised runtime guards. Local validation is 324 Rust tests and 2,487 Python tests with a rebuilt/source-verified extension.

File: passivbot-rust/src/backtest_hsl_revised.rs [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+                        continue;
+                    }
+                    if fill.index > k {
+                        return Err("future backtest HSL execution".into());
+                    }
+                    let sequence =
+                        i64::try_from(sequence).map_err(|_| "backtest HSL sequence overflow")?;
+                    fills.push(history::Fill {
+                        identity: sequence.to_string(),
+                        timestamp,
```
[the remark this answers is not in the document]
Comment (enarjord) [reply passivbot-rust/src/backtest_hsl_revised.rs:116]: Fixed in e8e8e84b806e51e3b100441c1b4f504889b7fb52. The adapter now declares explicit fill-after-same-timestamp-historical-price phase. Reconstruction and scoped trace consumption both honor it, including flatten rows; the current endpoint includes all observed fills. Timestamp labels remain unchanged. Long and short regressions verify an intrabar entry never acquires the preceding close's fictional profit or a false subsequent RED.

File: passivbot-rust/src/backtest_hsl_revised.rs [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+                        delta: Some(fill.fill_qty),
+                        price: Some(fill.fill_price),
+                        realized: Some(fill.pnl),
+                        fee: Some(fill.fee_paid),
+                        sequence: Some(sequence),
```
[the remark this answers is not in the document]
Comment (enarjord) [reply passivbot-rust/src/backtest_hsl_revised.rs:121]: Fixed in e8e8e84b806e51e3b100441c1b4f504889b7fb52. A separate global_fill_sequence producer contract now permits exact cross-pair cohort ordering. The simulator opts in; ordinary exchange inputs retain per-pair semantics. Real simulator-handler tests cover side and unified close/close/reopen; extension tests also verify that missing global authority or duplicate sequence values do not certify that boundary.

File: passivbot-rust/src/backtest_hsl_revised.rs [hunk 3]
```diff
@@ -0,0 +1,535 @@
...
+                        revision: 0,
+                    });
+                }
+                if position.size == 0.0 && fills.is_empty() && !self.coin_is_valid_at(idx, k) {
+                    // A pre-listing/expired flat pair has no scope contribution.
+                    continue;
```
[the remark this answers is not in the document]
Comment (enarjord) [reply passivbot-rust/src/backtest_hsl_revised.rs:137]: Fixed in e8e8e84b806e51e3b100441c1b4f504889b7fb52. An explicitly requested pre-listing/expired flat coin with no retained fills supplies a fresh coin/side/window-bound flat observation. Shared snapshot/trace consumers return a neutral current observation without an invented quote. Bare absence, stale/mismatched proof and conflicting retained pair inputs fail explicitly. Real simulator and extension regressions cover these cases.

**Comment by enarjord** [comment]
@codex review

Please review current head e8e8e84b806e51e3b100441c1b4f504889b7fb52, including all three previous findings. The changes add explicit historical candle/fill phase, declared global simulator ordering across pairs, and bound fresh flat-coin observations without fabricated quotes. Shared primitive defaults preserve ordinary exchange semantics. Validation: 327 Rust tests, default-feature compilation, 2,542 Python tests, documentation checks and a rebuilt/source-verified extension. Revised runtime guards remain in place.

File: passivbot-rust/src/backtest_hsl_revised.rs [hunk 4]
```diff
@@ -0,0 +1,658 @@
...
+                    .ok_or("missing backtest HSL valuation history")?;
+                let mark = self.hlcvs_value(latest, idx, CLOSE);
+                if !mark.is_finite() || mark <= 0.0 {
+                    return Err("invalid current backtest HSL valuation".into());
+                }
+                let mut candles = Vec::new();
+                for j in first_row..=k {
+                    let end = close_time(j)?;
+                    let candle_start = end - 60_000;
+                    if candle_start < start || !self.coin_is_valid_at(idx, j) {
```
[the remark this answers is not in the document]
Comment (enarjord) [reply passivbot-rust/src/backtest_hsl_revised.rs]: Fixed in eaf571e3dcd3d685f01a97af462f61b6bf300c74. The adapter now retains the source row whose instantaneous 1m close is at the inclusive left edge. This follows the existing shared price/reference rule (coarse buckets still require their whole interval inside the window); no earlier price sample or fill is retained. Added long/short actual simulator regressions with an opening outside the window, a close exactly at start, and a favorable next-bar close. The reconstructed boundary UPNL remains zero rather than inventing profit and a later drawdown. Validation: 328 Rust tests and 2,542 Python tests with rebuilt/source-verified extension.

**Comment by enarjord** [comment]
@codex review

Please re-review current head eaf571e3dcd3d685f01a97af462f61b6bf300c74, especially the inclusive left-edge close fix for r4058656998. The source candle may open one minute before the window, but only its instantaneous close at the inclusive boundary is consumed, as already permitted by the shared projector/reference. Coarse interpolation and fill scope remain unchanged. Long/short simulator regressions added; 328 Rust and 2,542 Python tests pass with a source-verified rebuild.

**Comment by enarjord** [comment]
Author integration check: merge ready.

Head: eaf571e3dcd3d685f01a97af462f61b6bf300c74
Base and effective merge base: e9d05de5779fb2d9fbd80e456efff3abf0e85cde

Independent current-head clearance: https://github.com/enarjord/passivbot/pull/1772#issuecomment-5754148511. All four review findings have implemented regressions and are resolved after that clearance. Rust and Python 3.12/3.14 CI checks pass on this head. Author validation includes 328 Rust tests, default-feature compilation, 2,542 Python tests and rebuilt/source-verified extension parity. The worktree is clean and GitHub reports the PR mergeable. No revised trading path is activated by this change.

reviewed by Codex (author integration check)
